pytest_inmanta/plugin.py

```diff
@@ -261,19 +274,36 @@ def project_metadata(request: pytest.FixtureRequest) -> module.ProjectMetadata:
         )
     )
 
+    pip_index_urls: Sequence[str] = pip_index_url.resolve(request.config)
     modulepath = ["libs"]
     in_place = inm_mod_in_place.resolve(request.config)
     if in_place:
         modulepath.append(str(Path(CURDIR).parent))
 
-    return module.ProjectMetadata(
-        name="testcase",
-        description="Project for testcase",
-        repo=repos,
-        modulepath=modulepath,
-        downloadpath="libs",
-        install_mode=inm_install_mode.resolve(request.config).value,
-    )
+    if SUPPORTS_PROJECT_PIP_INDEX:
+        # On newer versions of core we set the pip.index_url of the project.yml file
+        pip_config: ProjectPipConfig = ProjectPipConfig(index_url=list(pip_index_urls))
```

Comment:
We should ensure backwards compatibility in the other direction as well: if the user still uses the deprecated option we should supply it to `pip.index_url`.